CMakeLists.txt

```diff
@@ -1,330 +1,712 @@
-        )
-
-add_subdirectory (api)
-
-if (BUILD_PYTHON)
-  find_package (PythonInterp)
-  find_package (PythonLibs)
-  find_package (NumPy)
-endif ()
-
-if (PYTHONINTERP_FOUND)
-  execute_process ( COMMAND ${PYTHON_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(plat_specific=True, prefix='${CMAKE_INSTALL_PREFIX}'))"
-                    OUTPUT_VARIABLE _ABS_PYTHON_MODULE_PATH
-                    OUTPUT_STRIP_TRAILING_WHITESPACE )
-
-  get_filename_component (_ABS_PYTHON_MODULE_PATH ${_ABS_PYTHON_MODULE_PATH} ABSOLUTE)
-  file (RELATIVE_PATH _REL_PYTHON_MODULE_PATH ${CMAKE_INSTALL_PREFIX} ${_ABS_PYTHON_MODULE_PATH})
-
-  set (PYTHON_MODULE_PATH ${_REL_PYTHON_MODULE_PATH})
-
-endif ()
-
-if (BUILD_GUILE)
-  find_package (Guile)
-endif ()
-
-if (USE_SWIG)
-  find_package (SWIG)
-endif ()
-
-add_subdirectory (swig)
-
-if (BUILD_OCTAVE)
-  find_package (Octave)
-endif ()
-
-if (BUILD_MATLAB)
-  find_package (Matlab)
-endif ()
-
-if (OCTAVE_FOUND OR Matlab_FOUND)
-  add_subdirectory (octave)
-endif ()
-
-enable_testing ()
-add_subdirectory (test)
-
-set (CPACK_PACKAGE_NAME          "${CMAKE_PROJECT_NAME}")
-set (CPACK_PACKAGE_VERSION_MAJOR "${NLOPT_MAJOR_VERSION}")
-set (CPACK_PACKAGE_VERSION_MINOR "${NLOPT_MINOR_VERSION}")
-set (CPACK_PACKAGE_VERSION_PATCH "${NLOPT_BUGFIX_VERSION}")
-set (CPACK_SOURCE_GENERATOR      "TBZ2;TGZ"    )
-set (CPACK_BINARY_STGZ           "OFF"        )
-set (CPACK_BINARY_TBZ2           "ON"         )
-set (CPACK_BINARY_TGZ            "ON"         )
-set (CPACK_BINARY_TZ             "OFF"        )
-set (CPACK_SOURCE_IGNORE_FILES ".git;/build;.*~;${CPACK_SOURCE_IGNORE_FILES}")
-
-set (CPACK_SOURCE_PACKAGE_FILE_NAME ${CPACK_PACKAGE_NAME}-${NLOPT_VERSION_STRING})
-
-include (CPack)
-
-# configuration files
-export (TARGETS ${nlopt_lib} FILE ${PROJECT_BINARY_DIR}/NLoptLibraryDepends.cmake)
-
-# Install the export set for use with the install-tree
-install(EXPORT NLoptLibraryDepends
-        DESTINATION ${RELATIVE_INSTALL_CMAKE_DIR}
-        COMPONENT Development)
-
-# Create a NLOPTConfig.cmake file for the use from the install tree
-# and install it
-set (NLOPT_LIBRARIES ${nlopt_lib})
-
-set (NLOPT_CMAKE_DIR "${INSTALL_CMAKE_DIR}")
-file (RELATIVE_PATH rel_include_dir "${NLOPT_CMAKE_DIR}" "${INSTALL_INCLUDE_DIR}")
-list (APPEND RELATIVE_NLOPT_INCLUDE_DIRS ${rel_include_dir})
-
-file (RELATIVE_PATH rel_lib_dir "${NLOPT_CMAKE_DIR}" "${INSTALL_LIB_DIR}")
-list (APPEND RELATIVE_NLOPT_LIB_DIR ${rel_lib_dir})
-
-configure_file (cmake/NLoptConfig.cmake.in NLoptConfig.cmake @ONLY)
-configure_file (cmake/NLoptConfigVersion.cmake.in NLoptConfigVersion.cmake @ONLY)
-install (FILES
-          ${CMAKE_CURRENT_BINARY_DIR}/NLoptConfig.cmake
-          ${CMAKE_CURRENT_BINARY_DIR}/NLoptConfigVersion.cmake
-         DESTINATION ${RELATIVE_INSTALL_CMAKE_DIR}
-         COMPONENT Development)
-
+#==============================================================================
```

Comment:
Right, I did change CRLF to LF: https://github.com/stevengj/nlopt/pull/83/commits/160a105f6940f359725696ad335337d78482ccd8
I should have done it in a separate commit, sorry about that!

One [tip](https://github.com/blog/967-github-secrets) I recently learned is to add `?w=1` to the Github URL to ignore whitespace changes in diffs: https://github.com/stevengj/nlopt/pull/83/commits/160a105f6940f359725696ad335337d78482ccd8?diff=split&w=1